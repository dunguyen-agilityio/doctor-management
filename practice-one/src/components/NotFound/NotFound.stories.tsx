import { View } from 'react-native';

import type { Meta, StoryObj } from '@storybook/react';

import NotFound from './index';

const meta = {
  title: 'NotFound',
  component: NotFound,
  decorators: [
    (Story) => (
      <View style={{ flex: 1 }}>
        <Story />
      </View>
    ),
  ],
} satisfies Meta<typeof NotFound>;

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
