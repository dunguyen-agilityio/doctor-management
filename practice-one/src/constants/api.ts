import Constants from 'expo-constants';

export const API_ENTITIES = {
  FOODS: 'foods',
  FAVORITES: 'favorites',
};

export const API_ENDPOINT = Constants.expoConfig?.extra?.apiEndpoint;
