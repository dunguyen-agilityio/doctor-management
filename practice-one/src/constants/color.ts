export const COLOR = {
  PRIMARY: '#5c5c5c',
  WHITE: '#fff',
  PINK: '#91c788',
  LIGHT_GRAY: '#ececec',
  GRAY: '#7e7e7e',
  RED: '#ea1502',
  PURPLE: '#9059a7',
  BLACK: '#000000',
  LIGHT_BLACK: '#330600',
  WARNING: '#da6317',
  LIGHT_PURPLE: '#fff3f1',
  GREEN: '#27ae60',
  ORANGE: '#d89e4a',
  YELLOW: '#f2c94c',
  LIGHT_GREEN_10: 'rgba(28, 195, 121, 0.1)',
  DARK_GREEN: '#1cc379',
  SECONDARY: '#ff8473',
  BLACK_BEAN: '#330600',
  LIGHT_GREEN: '#91c788',
  GREY: '#e0e0e0',
  TERTIARY: '#696969',
};
