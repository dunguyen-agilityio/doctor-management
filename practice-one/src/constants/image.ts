export const BLUR_HASH =
  '|rF?hV%2WCj[ayj[a|j[az_NaeWBj@ayfRayfQfQM{M|azj[azf6fQfQfQIpWXofj[ayj[j[fQayWCoeoeaya}j[ayfQa{oLj?j[WVj[ayayj[fQoff7azayj[ayj[j[ayofayayayj[fQj[ayayj[ayfjj[j[ayjuayj[';

export enum APP_ICONS {
  EMPTY = require('@assets/images/empty.webp'),
  NOT_FOUND = require('@assets/images/not-found.webp'),
  LOGO = require('@assets/images/logo.webp'),
  ARROW_RIGHT_BOLD = require('@assets/icons/arrow-right-bold.webp'),
  ARROW_LEFT = require('@assets/icons/arrow-left.webp'),
  QUESTION = require('@assets/icons/question.webp'),
  SEARCH = require('@assets/icons/icon-search.webp'),
  FAVORITE = require('@assets/icons/favorite.webp'),
  FAVORITE_FILL = require('@assets/icons/favorite-fill.webp'),
  HOME = require('@assets/icons/home.webp'),
  HOME_FILL = require('@assets/icons/home-fill.webp'),
  SEARCH_MENU = require('@assets/icons/search.webp'),
  SEARCH_MENU_FILL = require('@assets/icons/search-fill.webp'),
}
