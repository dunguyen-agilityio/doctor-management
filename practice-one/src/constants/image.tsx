import { Image, StyleSheet } from 'react-native';

export const EmptyImage = () => (
  <Image
    source={require('@/assets/images/empty_2x.png')}
    style={styles.empty}
  />
);

export const NotFoundImage = () => (
  <Image
    source={require('@/assets/images/not_found_2x.png')}
    style={styles.notFound}
    testID="not-found-image"
  />
);

export const Logo = () => (
  <Image source={require('@/assets/images/logo.png')} style={styles.loading} />
);

export const ArrowICon = () => (
  <Image source={require('@/assets/icons/arrow.png')} />
);

export const QuestionIcon = () => (
  <Image source={require('@/assets/icons/question.png')} />
);

export const SearchIcon = () => (
  <Image source={require('@/assets/icons/iconsearch.png')} />
);

export const FavoriteMenu = ({ isFill = false }: { isFill?: boolean }) => {
  const path = isFill
    ? require('@/assets/icons/favoritefill.png')
    : require('@/assets/icons/favorite.png');
  return <Image source={path} />;
};

export const HomeMenu = ({ isFill = false }: { isFill?: boolean }) => {
  const path = isFill
    ? require('@/assets/icons/homefill.png')
    : require('@/assets/icons/home.png');
  return <Image source={path} />;
};

export const SearchMenu = ({ isFill = false }: { isFill?: boolean }) => {
  const path = isFill
    ? require('@/assets/icons/searchfill.png')
    : require('@/assets/icons/search.png');
  return <Image source={path} />;
};

export const SplashImage = () => (
  <Image
    source={require('@/assets/images/pattern.png')}
    style={styles.topImage}
  />
);

export const styles = StyleSheet.create({
  empty: {
    height: 96,
    width: 108,
  },
  notFound: {
    width: 96,
    height: 96,
  },
  loading: {
    width: 122,
    height: 122,
  },
  topImage: {
    width: '100%',
    resizeMode: 'cover',
  },
});
