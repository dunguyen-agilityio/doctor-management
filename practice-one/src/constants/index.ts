export * from './api';
export * from './color';
export * from './category';
export * from './image';
export * from './screens';
