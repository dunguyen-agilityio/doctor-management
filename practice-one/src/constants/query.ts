export const QUERY_KEYS = {
  FOOD: 'food-list',
  FOOD_FAVORITE: 'food-list-favorite',
  FOOD_BY_ID: (id: string) => `food-${id}`,
};

export const PAGE_SIZE = 10;
