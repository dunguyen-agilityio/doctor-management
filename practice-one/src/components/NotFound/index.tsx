import { StyleSheet, View } from 'react-native';

import { COLOR } from '@/constants';

import Text from '../Text';
import { NotFoundImage } from '../icons';

const NO_FOODS_FOUND_DESCRIPTION = `You don't save any food. Go ahead, search\nand save your favorite food`;
const NO_FOODS_FOUND_TITLE = 'No Foods Found';

const NotFound = ({
  description = NO_FOODS_FOUND_DESCRIPTION,
  image = <NotFoundImage />,
  title = NO_FOODS_FOUND_TITLE,
}: {
  title?: string;
  description?: string;
  image?: React.ReactNode;
}) => {
  return (
    <View testID="not-found-container" style={styles.container}>
      <View testID="not-found-header" style={styles.header}>
        {image}
        <Text variant="subtitle2" style={styles.title} color={COLOR.SECONDARY}>
          {title}
        </Text>
      </View>

      <Text variant="body1" color={COLOR.GRAY}>
        {description}
      </Text>
    </View>
  );
};

export default NotFound;

const styles = StyleSheet.create({
  container: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
    gap: 8,
    width: '100%',
  },
  header: {
    gap: 24,
    alignItems: 'center',
  },
  title: {
    marginTop: 24,
  },
  description: {
    textAlign: 'center',
  },
});
