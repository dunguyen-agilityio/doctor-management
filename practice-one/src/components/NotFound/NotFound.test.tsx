import React from 'react';
import { Image, Text } from 'react-native';

import { render } from '@testing-library/react-native';

import { COLORS } from '@/constants';

import NotFound from './index';

describe('NotFound Component', () => {
  const defaultTitle = 'No Foods Found';
  const defaultDescription = `You don't save any food. Go ahead, search\nand save your favorite food`;

  it('renders correctly with default props', () => {
    const { getByText, getByTestId } = render(<NotFound />);

    // Check if the default title is rendered
    expect(getByText(defaultTitle)).toBeTruthy();

    // Check if the default description is rendered
    expect(getByText(defaultDescription)).toBeTruthy();

    // Check if the container and header styles are applied
    const container = getByTestId('not-found-container');
    expect(container.props.style).toEqual({
      flex: 1,
      alignItems: 'center',
      justifyContent: 'center',
      gap: 8,
      width: '100%',
    });

    const header = getByTestId('not-found-header');
    expect(header.props.style).toEqual({
      gap: 24,
      alignItems: 'center',
    });

    // Check if the title and description styles are applied
    const title = getByText(defaultTitle);
    expect(title.props.style).toEqual({
      marginTop: 24,
      color: '#696969',
      fontSize: 22,
      fontWeight: '500',
      lineHeight: 31,
    });

    const description = getByText(defaultDescription);
    expect(description.props.style).toEqual({
      fontSize: 13,
      color: COLORS.GRAY,
      textAlign: 'center',
    });
  });

  it('renders correctly with custom props', () => {
    const customTitle = 'Custom Title';
    const customDescription = 'Custom Description';
    const CustomImage = () => <Image testID="not-found-image" />;

    const { getByText, getByTestId } = render(
      <NotFound
        title={customTitle}
        description={customDescription}
        image={<CustomImage />}
      />,
    );

    // Check if the custom title, description, and image are rendered
    expect(getByText(customTitle)).toBeTruthy();
    expect(getByText(customDescription)).toBeTruthy();
    expect(getByTestId('not-found-image')).toBeTruthy();
  });

  it('renders the default image when no custom image is provided', () => {
    const { getByTestId } = render(<NotFound />);

    // Check if the default image is rendered
    const defaultImage = getByTestId('not-found-image');
    expect(defaultImage).toBeTruthy();
  });
});
