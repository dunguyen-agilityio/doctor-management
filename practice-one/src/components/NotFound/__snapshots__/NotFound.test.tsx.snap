// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Component renders correctly with default props 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "flex": 1,
      "gap": 8,
      "justifyContent": "center",
      "width": "100%",
    }
  }
  testID="not-found-container"
>
  <View
    style={
      {
        "alignItems": "center",
        "gap": 24,
      }
    }
    testID="not-found-header"
  >
    <Image
      source={
        {
          "testUri": "../../../assets/images/not_found_2x.png",
        }
      }
      style={
        {
          "height": 96,
          "width": 96,
        }
      }
      testID="not-found-image"
    />
    <Text
      style={
        [
          {
            "color": "#ff8473",
            "fontFamily": "Manrope",
            "fontSize": 15,
            "fontWeight": "600",
            "lineHeight": 23,
            "textTransform": undefined,
          },
          {
            "marginTop": 24,
          },
        ]
      }
    >
      No Foods Found
    </Text>
  </View>
  <Text
    style={
      [
        {
          "color": "#5c5c5c",
          "fontFamily": "Manrope",
          "fontSize": 13,
          "fontWeight": "400",
          "lineHeight": 18,
          "textTransform": undefined,
        },
        undefined,
      ]
    }
  >
    You don't save any food. Go ahead, search
and save your favorite food
  </Text>
</View>
`;
